src/main/java/gregtech/api/block/VariantActiveBlock.java

```diff
@@ -1,44 +1,72 @@
 import gregtech.api.util.GTUtility;
+import gregtech.client.model.IModelSupplier;
+import gregtech.client.model.SimpleStateMapper;
+import gregtech.client.utils.BloomEffectUtil;
+import it.unimi.dsi.fastutil.objects.Object2BooleanOpenHashMap;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+import it.unimi.dsi.fastutil.objects.ObjectOpenHashSet;
+import it.unimi.dsi.fastutil.objects.ObjectSet;
 import net.minecraft.block.material.Material;
-import net.minecraft.block.properties.PropertyBool;
+import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyEnum;
 import net.minecraft.block.state.BlockStateContainer;
 import net.minecraft.block.state.IBlockState;
-import net.minecraft.item.ItemStack;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.client.renderer.block.statemap.IStateMapper;
+import net.minecraft.item.Item;
+import net.minecraft.util.BlockRenderLayer;
 import net.minecraft.util.IStringSerializable;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockAccess;
+import net.minecraftforge.client.event.TextureStitchEvent;
+import net.minecraftforge.client.model.ModelLoader;
+import net.minecraftforge.common.property.ExtendedBlockState;
+import net.minecraftforge.common.property.IExtendedBlockState;
+import net.minecraftforge.common.property.IUnlistedProperty;
+import net.minecraftforge.fml.common.Loader;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import team.chisel.ctm.client.state.CTMExtendedState;
 
 import javax.annotation.Nonnull;
 
-public class VariantActiveBlock<T extends Enum<T> & IStringSerializable> extends VariantBlock<T>{
-    public static final PropertyBool ACTIVE = PropertyBool.create("active");
+import static gregtech.common.blocks.MetaBlocks.statePropertiesToString;
+
+public class VariantActiveBlock<T extends Enum<T> & IStringSerializable> extends VariantBlock<T> implements IModelSupplier {
+
+    public static final ModelResourceLocation MODEL_LOCATION = new ModelResourceLocation(new ResourceLocation(GTValues.MODID, "active_blocks"), "inventory");
+    public static final IStateMapper mapper = new SimpleStateMapper(MODEL_LOCATION);
+    public static final Object2ObjectOpenHashMap<Integer, ObjectSet<BlockPos>> ACTIVE_BLOCKS = new Object2ObjectOpenHashMap<>();
+    public static final UnlistedBooleanProperty ACTIVE = new UnlistedBooleanProperty("active");
 
     public VariantActiveBlock(Material materialIn) {
         super(materialIn);
     }
 
     @Override
     public IBlockState getState(T variant) {
-        return super.getState(variant).withProperty(ACTIVE, false);
-    }
-
-    public IBlockState getState(T variant, boolean active) {
-        return super.getState(variant).withProperty(ACTIVE, active);
+        return super.getState(variant);
     }
```

Comment:
Is this method needed any more?

Reply:
this was used before to remove the active variants from the JEI

Reply:
I meant that all this method does now is call `super`, so couldn't it be removed?